v2/apiserver/internal/core/mongodb/events_store.go

```diff
@@ -368,15 +369,30 @@ func (e *eventsStore) CancelMany(
 			{Key: "created", Value: -1},
 		},
 	)
-	cur, err := e.collection.Find(ctx, criteria, findOptions)
+	count, err := e.collection.CountDocuments(ctx, criteria)
```

Comment:
We can get the count without making an extra database call.

In the one (or, conditionally, two) updates applied above, there's a result object we're currently ignoring that can give us a count of affected documents.